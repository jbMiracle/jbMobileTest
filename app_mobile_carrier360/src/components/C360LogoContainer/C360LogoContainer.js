import React, { PureComponent } from 'react';
import { Image, View } from 'react-native';
import { Container, Header, Text, H3, Content } from 'native-base';
import styleDef from './C360LogoContainer.style';
import C360Icon from '../C360Icon/C360Icon';

const logo360 = require('../../assets/images/FullLogo.png');

class C360LogoContainer extends PureComponent {
  render() {
    const {
      showBackButton, backButtonOnPress, headerText, descriptionText,
    } = this.props;
    return (
      <Container style={ styleDef.fillWidth }>
        {renderHeader(showBackButton, backButtonOnPress)}
        <Content style={ [styleDef.blueBackgroundColor, !showBackButton && styleDef.topPadding] }>
          <View style={ [styleDef.containerStyles, styleDef.horizontalPadding] }>
            <Image
              source={ logo360 }
              style={ styleDef.imageStyles }
            />
            { headerText ? <H3 style={ styleDef.headerTextStyles }> { headerText } </H3> : null}
            { descriptionText ? <Text style={ styleDef.descriptionTextStyles }> { descriptionText } </Text> : null}
            { this.props.children}
          </View>
        </Content>
      </Container>
    );
  }
}

function renderHeader(showBackButton, backButtonOnPress) {
  return (showBackButton &&
    <Header noShadow style={ [styleDef.headerStyles] } >
      <C360Icon name='jbh-Chevron_Back' color='white' size={ 20 } onPress={ backButtonOnPress } style={ styleDef.iconStyles } />
    </Header>
  );
}

export default C360LogoContainer;
