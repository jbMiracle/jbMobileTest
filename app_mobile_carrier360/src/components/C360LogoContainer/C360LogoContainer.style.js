import C360Colors from '../../themes/360Colors';
import variables from '../../themes/defaultTheme/variables/platform';

export default {
  containerStyles: {
    alignItems: 'center',
    backgroundColor: C360Colors.Blue500,
  },
  headerStyles: {
    backgroundColor: C360Colors.Blue500,
    borderBottomWidth: 0,
    flexDirection: 'column',
    alignItems: 'flex-start',
  },
  headerTextStyles: {
    color: C360Colors.White,
    textAlign: 'center',
  },
  descriptionTextStyles: {
    color: C360Colors.White,
    marginTop: 10,
    textAlign: 'center',
  },
  imageStyles: {
    marginBottom: 30,
  },
  iconStyles: {
    color: C360Colors.White,
  },
  blueBackgroundColor: {
    backgroundColor: C360Colors.Blue500,
  },
  topPadding: {
    paddingTop: variables.toolbarHeight,
  },
  horizontalPadding: {
    paddingHorizontal: 30,
  },
  fillWidth: {
    width: '100%',
  },
};
