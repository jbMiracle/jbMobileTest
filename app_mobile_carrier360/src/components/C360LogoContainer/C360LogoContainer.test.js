import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import C360LogoContainer from './C360LogoContainer';

it('can render an empty component', () => {
  const renderer = new ShallowRenderer();
  const props = {
    navigation: {
      navigate: () => {},
    },
  };

  renderer.render(<C360LogoContainer { ...props } />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it('renders the back button on the screen', () => {
  const renderer = new ShallowRenderer();

  const props = {
    showBackButton: true,
    backButtonOnPress: () => {},
  };

  renderer.render(<C360LogoContainer { ...props } />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it('can render an empty header without back button', () => {
  const renderer = new ShallowRenderer();

  const props = {
    showBackButton: false,
  };

  renderer.render(<C360LogoContainer { ...props } />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it('can render text with headerText', () => {
  const renderer = new ShallowRenderer();

  const props = {
    headerText: 'Thank you!',
  };

  renderer.render(<C360LogoContainer { ...props } />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

