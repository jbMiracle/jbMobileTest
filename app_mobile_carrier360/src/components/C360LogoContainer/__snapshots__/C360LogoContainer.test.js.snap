// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render an empty component 1`] = `
<Styled(Container)
  style={
    Object {
      "width": "100%",
    }
  }
  virtual={undefined}
>
  <Styled(Content)
    style={
      Array [
        Object {
          "backgroundColor": "#007DBA",
        },
        Object {
          "paddingTop": 64,
        },
      ]
    }
    virtual={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#007DBA",
          },
          Object {
            "paddingHorizontal": 30,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/FullLogo.png",
          }
        }
        style={
          Object {
            "marginBottom": 30,
          }
        }
      />
    </View>
  </Styled(Content)>
</Styled(Container)>
`;

exports[`can render an empty header without back button 1`] = `
<Styled(Container)
  style={
    Object {
      "width": "100%",
    }
  }
  virtual={undefined}
>
  <Styled(Content)
    style={
      Array [
        Object {
          "backgroundColor": "#007DBA",
        },
        Object {
          "paddingTop": 64,
        },
      ]
    }
    virtual={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#007DBA",
          },
          Object {
            "paddingHorizontal": 30,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/FullLogo.png",
          }
        }
        style={
          Object {
            "marginBottom": 30,
          }
        }
      />
    </View>
  </Styled(Content)>
</Styled(Container)>
`;

exports[`can render text with headerText 1`] = `
<Styled(Container)
  style={
    Object {
      "width": "100%",
    }
  }
  virtual={undefined}
>
  <Styled(Content)
    style={
      Array [
        Object {
          "backgroundColor": "#007DBA",
        },
        Object {
          "paddingTop": 64,
        },
      ]
    }
    virtual={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#007DBA",
          },
          Object {
            "paddingHorizontal": 30,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/FullLogo.png",
          }
        }
        style={
          Object {
            "marginBottom": 30,
          }
        }
      />
      <Styled(H3)
        style={
          Object {
            "color": "#FFFFFF",
            "textAlign": "center",
          }
        }
        virtual={undefined}
      >
         
        Thank you!
         
      </Styled(H3)>
    </View>
  </Styled(Content)>
</Styled(Container)>
`;

exports[`renders the back button on the screen 1`] = `
<Styled(Container)
  style={
    Object {
      "width": "100%",
    }
  }
  virtual={undefined}
>
  <Styled(Header)
    noShadow={true}
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#007DBA",
          "borderBottomWidth": 0,
          "flexDirection": "column",
        },
      ]
    }
    virtual={undefined}
  >
    <Icon
      allowFontScaling={false}
      color="white"
      name="jbh-Chevron_Back"
      onPress={[Function]}
      size={20}
      style={
        Object {
          "color": "#FFFFFF",
        }
      }
    />
  </Styled(Header)>
  <Styled(Content)
    style={
      Array [
        Object {
          "backgroundColor": "#007DBA",
        },
        false,
      ]
    }
    virtual={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#007DBA",
          },
          Object {
            "paddingHorizontal": 30,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/FullLogo.png",
          }
        }
        style={
          Object {
            "marginBottom": 30,
          }
        }
      />
    </View>
  </Styled(Content)>
</Styled(Container)>
`;
